pages/datasets/_datasetId.vue

```diff
@@ -307,12 +307,12 @@ const getThumbnailData = async (datasetDoi, datasetId, datasetVersion, datasetFa
         if (scicrunchData.organs[0]) { // Check if dataset has organ annotation
           let data = {}
           // If we do not have a flatmap for the species, check the rat flatmap
-          if (!taxo) {
-            taxo = Uberons.species['rat']
+          if (taxo) {
+            // if we do ahve a flatmap for the species, check if it has the anatomy annotated
             let anatomy = scicrunchData.organs.map(organ => organ.curie)
             data = await flatmaps.anatomyQuery(taxo, anatomy)
           } else {
-            // if we do ahve a flatmap for the species, check if it has the anatomy annotated
+            taxo = Uberons.species['rat']
             let anatomy = scicrunchData.organs.map(organ => organ.curie)
             data = await flatmaps.anatomyQuery(taxo, anatomy)
           }
```

Comment:
The last two lines of this if block are repeated and should be outside the if statement, you actually only need the `taxo = Uberons.species['rat']` apologies for not spotting that the first time.

Also, `anatomy` should be `const` because you are not modifying it.